lnwallet/channel.go

```diff
@@ -2506,32 +2511,92 @@ func NewBreachRetribution(chanState *channeldb.OpenChannel, stateNum uint64,
 	// If our balance exceeds the remote party's dust limit, instantiate
 	// the sign descriptor for our output.
 	if ourAmt >= int64(chanState.RemoteChanCfg.DustLimit) {
+		// As we're about to sweep our own output w/o a delay, we'll
```

Comment:
me gusta, tacked it on

Reply:
On second thought, I think passing through the single and double tweak ends up leaking a bit too much information, making the interface less generalized. To fix that we could pass in the entire commitment key ring, but then it would be bound more directly to the channel state machine itself.